pandas/core/internals/managers.py

```diff
@@ -1087,7 +1087,7 @@ def value_getitem(placement):
             unfit_mgr_locs = np.concatenate(unfit_mgr_locs)
             unfit_count = len(unfit_mgr_locs)
 
-            new_blocks = []
+            new_blocks: List[Block] = []
```

Comment:
rather than add the variable annotation, does new_blocks need to be initialized outside the if else? make_block has the correct return type and the extend wouldn't be needed. (but I guess this mypy fixup is orthogonal to this PR)

Reply:
I'm not a fan of type annotations when not required. but not a blocker.